src/util/rax_extensions.c

```diff
@@ -54,3 +54,17 @@ void **raxValues(rax *rax) {
 	return values;
 }
 
+unsigned char **raxKeys(rax *rax) {
+	// Instantiate an array to hold all of the keys in the rax.
+	unsigned char **keys = array_new(unsigned char *, raxSize(rax));
+	raxIterator it;
+	raxStart(&it, rax);
+	// Iterate over all keys in the rax.
+	raxSeek(&it, "^", NULL, 0);
+	while(raxNext(&it)) {
+		// Copy the key into the array.
+		keys = array_append(keys, (unsigned char *)rm_strndup((const char *)it.key, (int)it.key_len));
```

Comment:
Is `it.key` NULL terminated?
the caller should assume each string in `keys` is NULL terminated.

Reply:
I used it before in one iteration of this PR where I needed rax keys
If you want I can remove it